pkg/filesystem/subvolume.go

```diff
@@ -52,6 +57,66 @@ func List(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace
 	listCephFSSubvolumes(ctx, clientsets, operatorNamespace, clusterNamespace, includeStaleOnly, subvolumeNames)
 }
 
+// checkForExternalStorage checks if the external mode is enabled.
+func checkForExternalStorage(ctx context.Context, clientsets *k8sutil.Clientsets, clusterNamespace string) bool {
+	enable := false
+	cephclusters, err := clientsets.Rook.CephV1().CephClusters(clusterNamespace).List(ctx, v1.ListOptions{})
+	if err != nil {
+		logging.Fatal(fmt.Errorf("failed to list CephClusters: %q", err))
+	}
+	for _, cephcluster := range cephclusters.Items {
```

Comment:
```suggestion
for i := range cephclusters.Items {
		enable = cephclusters.Items[i].Spec.External.Enable
		if enable == true {
			break
		}
	}
	return enable
```

avoid in-memory copy when doing for loop